File: src/HMS.Domain/Entities/Medicine.cs
using HMS.Domain.Common;

namespace HMS.Domain.Entities;

public class Medicine : BaseEntity
{
    public string Name { get; set; } = string.Empty;
    public string? GenericName { get; set; }
    public string? Manufacturer { get; set; }
    public string Unit { get; set; } = string.Empty;
    public decimal UnitPrice { get; set; }
    public int StockQuantity { get; set; }
    public int ReorderLevel { get; set; } = 10;
    public DateTime? ExpiryDate { get; set; }

    public ICollection<PrescriptionItem> PrescriptionItems { get; set; } = new List<PrescriptionItem>();
}

File: frontend/src/types/medicine.ts
export interface MedicineDto {
  id: string;
  name: string;
  genericName: string | null;
  manufacturer: string | null;
  unit: string;
  unitPrice: number;
  stockQuantity: number;
  reorderLevel: number;
  isLowStock: boolean;
  expiryDate: string | null;
  createdAt: string;
}

export interface CreateMedicineDto {
  name: string;
  genericName?: string;
  manufacturer?: string;
  unit: string;
  unitPrice: number;
  stockQuantity: number;
  reorderLevel: number;
  expiryDate?: string;
}

export interface UpdateMedicineDto {
  name: string;
  genericName?: string;
  manufacturer?: string;
  unit: string;
  unitPrice: number;
  reorderLevel: number;
  expiryDate?: string;
}
